src/Extractor/EntityExtractor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\Hal\Extractor;
+
+use JsonSerializable;
+use SplObjectStorage;
+use Zend\Stdlib\Extractor\ExtractionInterface;
+use ZF\Hal\EntityHydratorManager;
+
+class EntityExtractor implements ExtractionInterface
+{
+    /**
+     * @var EntityHydratorManager
+     */
+    protected $entityHydratorManager;
+
+    /**
+     * Map of entities to their ZF\Hal\Entity serializations
+     *
+     * @var SplObjectStorage
+     */
+    protected $serializedEntities;
+
+    /**
+     * @param EntityHydratorManager $entityHydratorManager
+     */
+    public function __construct(EntityHydratorManager $entityHydratorManager)
+    {
+        $this->entityHydratorManager = $entityHydratorManager;
+        $this->serializedEntities    = new SplObjectStorage();
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Should be `@inheritDoc` without the braces if you want to inherit the whole thing.